server/src/main/java/com/flipkart/varadhi/auth/DefaultAuthorizationProvider.java

```diff
@@ -127,27 +130,25 @@ private List<Pair<ResourceType, ResourceContext>> generateResourceContextHierarc
     private boolean isAuthorizedInternal(
             String subject, ResourceAction action, ResourceContext resourceContext
     ) {
-        log.debug(
-                "Checking authorization for subject [{}] and action [{}] on resource [{}]", subject, action,
-                resourceContext.policyPath()
-        );
         return getRolesForSubject(subject, resourceContext)
                 .stream().anyMatch(role -> doesActionBelongToRole(subject, role, action));
     }
 
     private Set<String> getRolesForSubject(String subject, ResourceContext resourceContext) {
-        RoleBindingNode node =
-                getAuthZService().getIAMPolicy(resourceContext.resourceType(), resourceContext.resourceId());
-        if (node == null) {
-            log.error("No roles on resource for subject {}", subject);
+        try {
```

Comment:
wrapped in try-catch so that exception thrown by getIamPolicy does not expose internal error and only shows as authorization failure

Reply:
sure, only catching `ResourceNotFoundException` as it would indicate no IAM Policy was set for the resource. Which implies that no one is authorized yet.